### PR TITLE
[#1798] course view admission and resign (connects #1798)

### PR DESCRIPTION
**Pull request by singharpita**

### CODE REVIEW

File: src/app/courses/courses.component.ts [hunk 1]
```diff
@@ -259,17 +259,18 @@ export class CoursesComponent implements OnInit, AfterViewInit, OnDestroy {
...
-    this.updateShelf(userShelf, 'Course added to your dashboard');
+  courseToggle(courseId, type) {
+    if (type === 'resign') {
+      this.coursesService.courseResign(courseId).subscribe((res) => {
+        this.setupList(this.courses.data,  this.userShelf.courseIds);
+        this.planetMessageService.showMessage('Course successfully resigned from myCourses');
+      }, (error) => ((error)));
+    } else {
+      this.coursesService.courseAdmission(courseId).subscribe((res) => {
+        this.setupList(this.courses.data,  this.userShelf.courseIds);
```
Comment (paulbert) [inline src/app/courses/courses.component.ts]: Remove extra space after comma

File: src/app/courses/courses.service.ts [hunk 2]
```diff
@@ -35,6 +35,7 @@ export class CoursesService {
       this.updateCourses(this.createCourseList(this.courses, ratings.docs));
       this.updateCourse({ course: this.createCourseList([ this.course ], ratings.docs)[0], progress: this.progress });
     });
+    console.log(this.userService.getShelf());
```
Comment (paulbert) [inline src/app/courses/courses.service.ts]: Please remove console.log statements

File: src/app/courses/courses.component.ts [hunk 3]
```diff
@@ -259,17 +259,18 @@ export class CoursesComponent implements OnInit, AfterViewInit, OnDestroy {
...
-  }
-
-  courseAdmission(courseId) {
-    const userShelf: any = { courseIds: [ ...this.userShelf.courseIds ], ...this.userShelf };
-    userShelf.courseIds.push(courseId);
-    this.updateShelf(userShelf, 'Course added to your dashboard');
+  courseToggle(courseId, type) {
+    if (type === 'resign') {
+      this.coursesService.courseResign(courseId).subscribe((res) => {
+        this.setupList(this.courses.data,  this.userShelf.courseIds);
```
Comment (paulbert) [inline src/app/courses/courses.component.ts]: Remove extra space after comma

**Comment by singharpita** [comment]
updated the function by removing repetitive code.
courseToggle is present both in CourseComponent and CourseViewComponent since CourseService can't be called from Component.

**Pull request merged**